src/main/java/com/amazonaws/secretsmanager/sql/AWSSecretsManagerDriver.java

```diff
@@ -376,22 +376,23 @@ public Connection connect(String url, Properties info) throws SQLException {
         if (url.startsWith(SCHEME)) { // If this is a URL in the correct scheme, unwrap it
             unwrappedUrl = unwrapUrl(url);
         } else { // Else, assume this is a secret ID and try to retrieve it
-            String secretString = secretCache.getSecretString(url);
-            if (StringUtils.isNullOrEmpty(secretString)) {
-                throw new IllegalArgumentException("URL " + url + " is not a valid URL starting with scheme " +
-                        SCHEME + " or a valid retrievable secret ID ");
-            }
-
             try {
+                String secretString = secretCache.getSecretString(url);
+                if (StringUtils.isNullOrEmpty(secretString)) {
+                    throw new IllegalArgumentException("URL " + url + " is not a valid URL starting with scheme " +
+                            SCHEME + " or a valid retrievable secret ID ");
+                }
```

Comment:
I'm not sure this fixes the original issue.  If 'username' or 'password' is missing in the JSON, in lines [343-344](https://github.com/aws/aws-secretsmanager-jdbc/blob/master/src/main/java/com/amazonaws/secretsmanager/sql/AWSSecretsManagerDriver.java#L344), then JSON Node will return 'null', and we'll get a nullException, which isn't caught anywhere. 

We'll want to handle no username or password up there too.

Reply:
The lines you pointed to are within the private connectWithSecret method. This method is only invoked after the [secretCache.getSecretString(url)](https://github.com/aws/aws-secretsmanager-jdbc/blob/master/src/main/java/com/amazonaws/secretsmanager/sql/AWSSecretsManagerDriver.java#L379) call is made on Line 379. This call results in an NPE if a previously valid secret is modified outside of the driver flow to remove the username or password parameter (as described in the original issue) and never makes it to the connectWithSecret call on line 402.

Reply:
@danmancuso as we discussed offline, there appears to be 2 separate NPE opportunities here that need addressed:
1. In the public connect method if a secret's username and/or password _value_ has been removed 
2. In the private method if the username and/or password fields are removed entirely (removal of _key and value_)

Commit 1 addresses the first while commit 2 addresses the second.